### DISCUSSION TITLE
staleTime with NextJS On Demand ISR

### DISCUSSION THREAD

**Discussion by dottodot** [TanStack/query#4617] [General]
I having some issues getting React Query to play nice with NextJS On Demand ISR.

For my setup I using the hydration method.

If I have a staleTime greater than zero On Demand ISR doesn't work, so I assuming that when the page is revalidated it using the cached data so the page never updates. Also in this situation I don't see any requests for data on the client.

If I set staleTime to zero On Demand ISR works, i.e the page gets updated and is cached but I've noticed that the client is always making a request for data which seems like it defeats the purpose of serving a static page.

Does anyone have any suggestions on what the best way might be to overcome this?

**Comment by TkDodo** [comment]
ISR happens on the server, right? I don't see how this correlates to staleTime or client side fetching?

**Comment by dottodot** [comment]
OK think I've the solution, I was using a shared `QueryClient` for all requests, and I've noticed the docs recommend using a new one for each query. After changing to this everything works as it should.
